pkg/volume/csi/expander.go

```diff
@@ -94,12 +94,30 @@ func (c *csiPlugin) nodeExpandWithClient(
 		return false, nil
 	}
 
-	volumeTargetPath := resizeOptions.DeviceMountPath
+	pv := resizeOptions.VolumeSpec.PersistentVolume
+	if pv == nil {
+		return false, fmt.Errorf("Expander.NodeExpand failed to find associated PersistentVolume for plugin %s", c.GetPluginName())
+	}
+
+	opts := csiResizeOptions{
+		volumePath:        resizeOptions.DeviceMountPath,
+		stagingTargetPath: resizeOptions.DeviceStagePath,
+		volumeid:          csiSource.VolumeHandle,
+		newSize:           resizeOptions.NewSize,
+		fsType:            csiSource.FSType,
+		accessMode:        api.ReadWriteOnce,
+		mountOptions:      pv.Spec.MountOptions,
+	}
 	if !fsVolume {
-		volumeTargetPath = resizeOptions.DevicePath
+		opts.volumePath = resizeOptions.DevicePath
+		opts.fsType = fsTypeBlockName
+	}
+
+	if pv.Spec.AccessModes != nil {
+		opts.accessMode = pv.Spec.AccessModes[0]
```

Comment:
What if access modes defined but empty, or multiple access modes specified?

Reply:
This is same as `NodeStage` code flow, where you can only pass one access mode. Currently as per CSI spec, capabilities should mostly be used for determining if device is a raw block device or fstype volume and hence accessMode should matter less. But even if a Driver does decide to use this field then it should only matter how the device is staged on *this* node.